Add library release notes for update 9 (#8908)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.9.0%22+label%3AType%2FBug).

### Revamped library updates

> The following new connectors along with major updates have been added to the Ballerina library, as part of the Ballerina connector revamp initiative. 

Reviewer: Topic says library and this paragraph says connectors, so it is bit confusing. Shall we use consistent workding?

Author: Agreed. Updated the title as `Revamped connector updates` with https://github.com/ballerina-platform/ballerina-dev-website/pull/8908/commits/8f18beca8757960898400b0878a0218a3d6441f3.

All the connectors listed below have been released under new major versions, due to significant changes in the APIs and functionalities.

#### `asana` package

- Enhanced connector APIs by incorporating resource function syntax, along with improved documentation and examples.

#### `candid` package 

- Introduced support for connecting to Candid's Charity Check, Essentials and Premier REST APIs.

#### `redis` package

- Added support to connect and work with Redis clusters.
- Introduced support for secure connections (SSL/TLS) to Redis servers.
- Added support for Redis connection strings (i.e. Redis URIs).

> - Extended connector compatibility to include the latest Redis server versions (upto 7.2.x).

Reviewer: ```suggestion
- Extended connector compatibility to include the latest Redis server versions (up to 7.2.x).
```

Author: Updated with https://github.com/ballerina-platform/ballerina-dev-website/pull/8908/commits/8f18beca8757960898400b0878a0218a3d6441f3

- Introduced Avro serialization/deserialization support.

#### `aws.redshift` package

- Introduced seamless connectivity to Amazon Redshift databases.

> 

Reviewer 2: @aashikam

> 

Reviewer 3: https://github.com/ballerina-platform/ballerina-dev-website/pull/8930

To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.9.0%22+label%3AType%2FBug).

### Revamped connector updates

> The following new connectors along with major updates have been added to the Ballerina library, as part of the Ballerina connector revamp initiative.

Reviewer 4: Should we split these further to new connectors and revamps? May help the descriptions.

```suggestion
New versions of the following connectors have been released with major updates, as part of the Ballerina connector revamp initiative.
```

All listed connectors have been released under new major versions, featuring significant API and functionality changes, along with improved documentation and examples.

#### `asana` package

> - Enhanced connector APIs by incorporating resource function syntax.

Reviewer 4: ```suggestion
- Enhanced connector APIs by incorporating resource method syntax.
```

#### `candid` package

> - Introduced support for connecting to Candid's Charity Check, Essentials and Premier REST APIs.

Reviewer 4: ```suggestion
- Introduced support for connecting to Candid's Charity Check, Essentials, and Premier REST APIs.
```

#### `confluent.cavroserdes` package

- Introduced support for serializing and deserializing data using Avro schemas stored in the Confluent Schmema Registry.

#### `confluent.cregistry` package

- Introduced new APIs to connect with the Confluent Schema Registry.

#### `docusign.dsadmin` package

> - Enhanced the connector APIs by incorporating resource functions.

Reviewer 4: ```suggestion
- Enhanced the connector APIs by incorporating resource methods.
```

> - The package name has been changed to `docusign.dsadmin` from its previous name, `docusign.admin`.

Reviewer 4: ```suggestion
- The package name has been changed to `docusign.dsadmin` from `docusign.admin`.
```

#### `mongodb` package

- Introduced `mongodb:Client`, `mongodb:Database`, and `mongodb:Collection` objects to provide a more intuitive and user-friendly API for MongoDB operations.
- Introduced new APIs to be consistent with MongoDB's native APIs.
- Added support for MongoDB aggregation operations.
- Added support for MongoDB projections through type inference and manual projection.
- Added support for connecting to MongoDB Atlas databases and clusters.
- Added support for SSL connections to MongoDB servers.
- Improved the `find` API with support for filtering, sorting, and pagination.
- Improved the `update` API by providing `matchedCount` and `modifiedCount` in the response.

#### `redis` package

- Added support to connect and work with Redis clusters.
- Introduced support for secure connections (SSL/TLS) to Redis servers.
- Added support for Redis connection strings (i.e. Redis URIs).
- Extended connector compatibility to include the latest Redis server versions (up to 7.2.x).

#### `zendesk` package

- Introduced support to connect to Zendesk REST API V2, combining the functionalities of the `zendesk.support` and `zendesk.voice` packages.

#### `rabbitmq` package

> - Removed the previously deprecated `rabbitmq:Message` record. Consequently, corresponding APIs no longer accommodate this record. Users are advised to transition to utilizing subtypes of `rabbitmq:AnydataMessage` for continued functionality.

Author: +1 for switch. @aashikam FYA

- Enhanced connector APIs by incorporating resource function syntax, along with improved documentation and examples.

#### `twilio` package

- Enhanced integration with Twilio services (messaging, voice calls, media services, etc.) offered via the Twilio Basic REST API version 2010-04-01. 

#### `salesforce` package

- Introduced client credentials flow and username-password flow options to the Salesforce connector.
- Expanded support to include Bulk v2 API, APEX REST API, and additional REST API functionalities. 
- Updated compatibility with the latest Salesforce REST API version (v59).
- Improved documentation and examples.

#### `aws.redshift` package

- Introduced seamless connectivity to Amazon Redshift databases.

#### `snowflake` package

- Removed the `requestGeneratedKeys` option from the Client configuration. Given that Snowflake databases do not return generated keys, the `requestGeneratedKeys` option is now defaulted to `NONE`.

#### `aws.dynamodb` package

- Introduced new APIs to work with database backups with DynamoDB REST APIs.

#### `aws.dynamodbstreams` package

- Introduced a new connector to work with DynamoDB streams.

#### `aws.sns` package

> - Enhanced the connector APIs to support all the APIs provided by the AWS SNS REST API.

Reviewer 2: It would be better if we could organize the connectors alphabetically here. (It seemed this was ordered alphabetically previously, but now there are connectors that doesn't follow this order).

Author: @ThisaruGuruge agreed. I sent the initial PR with alphabetically ordered entries, which seems not preserved with the subsequent PRs. Reordered with https://github.com/ballerina-platform/ballerina-dev-website/pull/8908/commits/77bd17ff9ff93e45abec65497286d9218185a846

To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.9.0%22+label%3AType%2FBug).

### Revamped connectors

New versions of the following connectors have been released with major updates, as part of the Ballerina connector revamp initiative.
All listed connectors have been released under new major versions, featuring significant API and functionality changes, along with improved documentation and examples.

#### `asana` package

> - Enhanced connector APIs by incorporating resource method syntax.
> 
> 

Author: ```suggestion
- Enhanced connector APIs by incorporating resource method syntax.

```

#### `asana` package

- Enhanced connector APIs by incorporating resource method syntax.

#### `aws.dynamodb` package

- Introduced new APIs to work with database backups with DynamoDB REST APIs.

#### `aws.dynamodbstreams` package

- Introduced a new connector to work with DynamoDB streams.

#### `aws.redshift` package

- Introduced seamless connectivity to Amazon Redshift databases.

#### `aws.sns` package

- Enhanced the connector APIs to support all the APIs provided by the AWS SNS REST API.

#### `candid` package

- Introduced support for connecting to Candid's Charity Check, Essentials, and Premier REST APIs.

#### `confluent.cavroserdes` package

- Introduced support for serializing and deserializing data using Avro schemas stored in the Confluent Schmema Registry.

#### `confluent.cregistry` package

- Introduced new APIs to connect with the Confluent Schema Registry.

#### `docusign.dsadmin` package

- Enhanced the connector APIs by incorporating resource methods.
- The package name has been changed to `docusign.dsadmin` from `docusign.admin`.

#### `docusign.dsclick` package

> - Enhanced the connector APIs by incorporating resource functions.

Reviewer 4: ```suggestion
- Enhanced the connector APIs by incorporating resource methods.
```

Let's update wherever applicable.

Author: @Nuvindu FYA

Reviewer 5: Fixed in [e7bdffd](https://github.com/ballerina-platform/ballerina-dev-website/pull/8908/commits/e7bdffd2803e9b07909b83821893089661d08ca9)

> - The package name has been changed to `docusign.dsclick` from its previous name, `docusign.click`.

Reviewer 4: Shall we update these also similar to L511? Please check wherever applicable.

Reviewer 5: Fixed in [e7bdffd](https://github.com/ballerina-platform/ballerina-dev-website/pull/8908/commits/e7bdffd2803e9b07909b83821893089661d08ca9)

#### `nats` package

- Removed the previously deprecated `nats:Message` record. Consequently, corresponding APIs no longer accommodate this record. Users are advised to transition to utilizing subtypes of `nats:AnydataMessage` for continued functionality.

#### `googleapis.gmail` package

- Enhanced connector APIs by incorporating resource function syntax, along with improved documentation and examples.

#### `twilio` package

- Enhanced integration with Twilio services (messaging, voice calls, media services, etc.) offered via the Twilio Basic REST API version 2010-04-01. 

#### `salesforce` package

- Introduced client credentials flow and username-password flow options to the Salesforce connector.
- Expanded support to include Bulk v2 API, APEX REST API, and additional REST API functionalities. 
- Updated compatibility with the latest Salesforce REST API version (v59).
- Improved documentation and examples.

#### `snowflake` package

- Removed the `requestGeneratedKeys` option from the Client configuration. Given that Snowflake databases do not return generated keys, the `requestGeneratedKeys` option is now defaulted to `NONE`.

> 

Reviewer 6: ```suggestion
#### `aws.dynamodb` package

- Introduced new APIs to work with database backups with DynamoDB REST APIs.

#### `aws.dynamodbstreams` package

- Introduced a new connector to work with DynamoDB streams.

#### `aws.sns` package

- Enhanced the connector APIs to support all the APIs provided by the AWS SNS REST API.
```

#### `googleapis.gmail` package

- Enhanced connector APIs by incorporating resource function syntax, along with improved documentation and examples.

#### `twilio` package

- Enhanced integration with Twilio services (messaging, voice calls, media services, etc.) offered via the Twilio Basic REST API version 2010-04-01. 

#### `salesforce` package

- Introduced client credentials flow and username-password flow options to the Salesforce connector.
- Expanded support to include Bulk v2 API, APEX REST API, and additional REST API functionalities. 
- Updated compatibility with the latest Salesforce REST API version (v59).
- Improved documentation and examples.

#### `snowflake` package

- Removed the `requestGeneratedKeys` option from the Client configuration. Given that Snowflake databases do not return generated keys, the `requestGeneratedKeys` option is now defaulted to `NONE`.

#### `aws.dynamodb` package

- Introduced new APIs to work with database backups with DynamoDB REST APIs.

#### `aws.dynamodbstreams` package

- Introduced a new connector to work with DynamoDB streams.

#### `aws.sns` package

> - Enhanced the connector APIs to support all the APIs provided by the AWS SNS REST API.

Reviewer 6: ```suggestion
- Enhanced the connector APIs to support all the APIs provided by the AWS SNS REST API.

```

To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.9.0%22+label%3AType%2FBug).

### Revamped connector updates

The following new connectors along with major updates have been added to the Ballerina library, as part of the Ballerina connector revamp initiative.
All listed connectors have been released under new major versions, featuring significant API and functionality changes, along with improved documentation and examples.

#### `asana` package

> - Enhanced connector APIs by incorporating resource function syntax, along with improved documentation and examples.

Reviewer 2: This part is redundant.

```suggestion
- Enhanced connector APIs by incorporating resource function syntax.
```
